tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/KnowledgeGuardian.scala

```diff
@@ -17,65 +17,33 @@ import org.scalajs.core.ir.Trees.FieldDef
 import org.scalajs.core.tools.linker._
 
 private[emitter] object KnowledgeGuardian {
-  private[KnowledgeGuardian] trait Unregisterable {
+  trait Unregisterable {
     def unregister(invalidatable: Invalidatable): Unit
   }
 
   trait Invalidatable {
-    private[KnowledgeGuardian] def registeredTo(
-        unregisterable: Unregisterable): Unit
+    def registeredTo(unregisterable: Unregisterable): Unit
```

Comment:
I had to make this public in the version where `KnowledgeAccessor` is in the class `KnowledgeGuardian`. Otherwise scalac would complain with abstract methods not being implemented although there was no reason. Apparently refchecks got confused with `private[KnowledgeGuardian]`, thinking it means something different in the context of the object or the class :(